cmd/up.go

```diff
@@ -118,11 +123,29 @@ List of components that kubeapps up installs:
 			c.SkipGc = false
 		}
 
-		if err = c.Run(objs, wd); err != nil {
-			return fmt.Errorf("can't install kubeapps components: %v", err)
+		// mongodb secret
+		// FIXME (tuna): if the mongodb secret exists then do nothing,
+		// otherwise, add it (with new generated rand pw) to the objs list then do full update.
+		if prevsecret, exist, err := mongoSecretExists(c, MongoDB_Secret, Kubeapps_NS); err != nil {
+			return err
+		} else if !exist {
+			pwFields := []string{"mongodb-password", "mongodb-root-password"}
+			pw := make(map[string]string)
+			for _, p := range pwFields {
+				s, err := generateRandomString(10)
+				if err != nil {
+					return fmt.Errorf("secret %s can't be created due to the random generator get failed: %v", MongoDB_Secret, err)
+				}
+				pw[p] = s
```

Comment:
I think you need to base64 encode this again.

I think the password should be printable characters, which you get from the first base64.  Secret.data is always base64 encoded, hence the second encoding.

Reply:
Ah that's a confused function name. `generateRandomString` actually returns a standard base64 encoded generated random string :P I will update that name

Reply:
Have you actually tried running mongodb with one of these Secrets?

I think what you're doing here is generating a random byte string (containing non-printable 8bit characters) and then base64-encoding it _once_, and putting it in Secret.data.  This means that the mongodb pod that uses this will be given MONGODB_PASSWORD=<8bit binary data>.  Does nami/mongodb/javascript/etc actually accept all possible characters (including null and invalid utf8) in its passwords? (I strongly doubt it)

Repeating: I think you should add a second base64 encoding so that the first base64 is used to generate your actual password (ie: it is now composed only of the base64 character-set rather than binary bytes), and then the second base64 is for the Secret.data encoding.

Reply:
I tried to run it once and saw the mongo pod running, but probably the secret pw generated wasn't luckily a violated one. Ok I saw your point and I updated it. Thank you Gus 👍